docs/lib/index.rst

```diff
@@ -0,0 +1,221 @@
+offering first class typing support and a full `Pydantic <https://github.com/samuelcolvin/pydantic>`_
+integration.
+
+The Starlite framework supports :doc:`/lib/usage/plugins/index`, ships
+with :doc:`dependency injection </lib/usage/dependency-injection>`, :doc:`security primitives </lib/usage/security/index>`,
+:doc:`OpenAPI schema generation </lib/usage/openapi>`, `MessagePack <https://msgpack.org/>`_,
+:doc:`middlewares </lib/usage/middleware/index>` and much more.
+
+Installation
+------------
+
+.. code-block:: shell
+
+   pip install starlite
+
+
+.. dropdown:: Extras
+    :icon: star
+
+    :ref:`Brotli Compression Middleware <lib/usage/middleware/builtin-middleware:brotli>`:
+        :code:`pip install starlite[brotli]`
+
+    :ref:`Client-side sessions <lib/usage/middleware/builtin-middleware:client-side sessions>`
+        :code:`pip install starlite[cryptography]`
+
+    :ref:`Server-side sessions <lib/usage/middleware/builtin-middleware:redis storage>` / :ref:`redis caching <lib/usage/caching:redis>`:
+        :code:`pip install starlite[redis]`
+
+    :ref:`Server-side sessions <lib/usage/middleware/builtin-middleware:memcached storage>` / :ref:`redis caching <lib/usage/caching:memcached>`:
+        :code:`pip install starlite[memcached]`
+
+    :ref:`Picologging <lib/usage/the-starlite-app:using picologging>`
+        :code:`pip install starlite[picologging]`
+
+    :ref:`StructLog <lib/usage/the-starlite-app:using structlog>`
+        :code:`pip install starlite[structlog]`
+
+    :doc:`/lib/usage/contrib/open-telemetry`
+        :code:`pip install starlite[openetelemetry]`
+
+    :doc:`/lib/usage/cli`
+        :code:`pip install starlite[cli]`
+
+    Standard installation (includes CLI, picologging and Jinja2 templating):
+        :code:`pip install starlite[standard]`
+
+    All extras:
+        :code:`pip install starlite[full]`
+
+
+
+Minimal Example
+---------------
+
+**Define your data model** using pydantic or any library based on it (for example ormar, beanie, SQLModel):
+
+.. code-block:: python
+
+    from pydantic import BaseModel, UUID4
+
+
+    class User(BaseModel):
+        first_name: str
+        last_name: str
+        id: UUID4
+
+
+
+
+You can also use dataclasses (standard library and Pydantic),
+:class:`typing.TypedDict` or :class:`msgspec.Struct`.
+
+.. code-block:: python
+
+   from uuid import UUID
+
+   # from pydantic.dataclasses import dataclass
+   from dataclasses import dataclass
+
+
+   @dataclass
+   class User:
+       first_name: str
+       last_name: str
+       id: UUID
+
+**Define a Controller for your data model:**
+
+.. code-block:: python
+
+    from typing import List
+
+    from pydantic import UUID4
+    from starlite import Controller, Partial, get, post, put, patch, delete
+
+    from my_app.models import User
+
+
+    class UserController(Controller):
+        path = "/users"
+
+        @post()
+        async def create_user(self, data: User) -> User:
+            ...
```

Comment:
One complaint that was often repeated is that the basic examples are not runnable due to the use of ellipses. 

I suggest remaking this example- simpler and runnable. I understand of course if it's our if scope.